indico/modules/search/client/js/components/results/Contribution.jsx

```diff
@@ -13,44 +15,55 @@ import {toMoment, serializeDate} from 'indico/utils/date';
 
 import '../ResultList.module.scss';
 
-const Contribution = ({title, url, startDt, persons}) => (
-  <div styleName="item">
-    <List.Header styleName="header">
-      <a href={url}>{title}</a>
-    </List.Header>
-    <List.Description styleName="description">
-      <List.Item>
-        {persons.length !== 0 && (
-          <ul>
-            {persons.length > 1 ? <Icon name="users" /> : <Icon name="user" />}
-            {persons.map(person => (
-              <li key={person.id}>
-                {person.title ? `${person.title} ${person.name}` : person.name}
-              </li>
-            ))}
-          </ul>
-        )}
-      </List.Item>
-      {startDt && (
+const Contribution = ({eventId, contributionId, title, startDt, persons}) => {
+  persons = [...new Map(persons.map(p => [p.name + p.affiliation, p])).values()];
+
+  return (
+    <div styleName="item">
+      <List.Header styleName="header">
+        <a href={contributionURL({event_id: eventId, contrib_id: contributionId})}>{title}</a>
+      </List.Header>
+      <List.Description styleName="description">
         <List.Item>
-          <Icon name="calendar alternate outline" />
-          {serializeDate(toMoment(startDt), 'DD MMMM YYYY HH:mm')}
+          {persons.length !== 0 && (
+            <ul>
+              {persons.length > 1 ? <Icon name="users" /> : <Icon name="user" />}
+              {persons.map(person => (
+                <li key={person}>
+                  {person.name}
+                  {person.affiliation ? ` (${person.affiliation.substring(0, 12)})` : ''}
```

Comment:
Shouldn't we ellipsize this instead of just truncating?

Reply:
Well, we could but I noticed there were some really long affiliations such as University of Somewhere Really Far Away, and having all of those occupying so much space from other person's names is excessive.

Reply:
It looks a bit strange:

![image](https://user-images.githubusercontent.com/179599/115382663-3b4dea00-a1d5-11eb-9c25-b4e875a65b85.png)

but even without trailing space it's a bit strange if it's cut off in the middle

![image](https://user-images.githubusercontent.com/179599/115383032-a26b9e80-a1d5-11eb-9540-78818051636f.png)



Maybe we could just show the names there and put the affiliation in a tooltip? Alternatively, wrapping the affiliation in a span with a max width + text-overflow ellipsis could be an option as well.

But I'm not sure if showing the affiliation there immediately is that important: I just checked the old CERN Search and we don't show the affiliation there at all. So maybe for now we can just omit it and if someone asks for it we revisit this?